__tests__/app.test.js

```diff
@@ -113,6 +113,186 @@ describe("/api/articles", () => {
+        });
+    });
+
+    test("sort_by & order topic ASC 200: responds with an array of article objects sorted by topics and by ascending alphbetical order when specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=topic&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("topic");
+        });
+    });
+
+    test("sort_by & order author ASC 200: responds with an array of article objects sorted by authors and by ascending alphbetical order when specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("author");
+        });
+    });
+
+    test("sort_by & order votes ASC 200: responds with an array of article objects sorted by number of votes and by ascending  order when specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=votes&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("votes");
+        });
+    });
+
+    test("sort_by & order created_at ASC 200: responds with an array of article objects sorted by creation date by ascending  order when specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=created_at&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("created_at");
+        });
+    });
+
+    test("sort_by & order comment_count ASC 200: responds with an array of article objects sorted by number of comments by ascending  order when specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=comment_count&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("comment_count");
+        });
+    });
+
+    test("sort_by & order title ASC 200: responds with an array of article objects sorted alphabetically by title when asc specified", () => {
+      return request(app)
+        .get("/api/articles?sort_by=title&order=asc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("title");
+        });
+    });
+    test("sort_by & order 400: responds with a bad request error message when given an invalid order or sort by query", () => {
+      return request(app)
+        .get("/api/articles?sort_by=i-am-an-invalid-query&order=asc")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("bad request");
+        });
+    });
```

Comment:
I think it's possible here for someone to misspell "order" or "sort_by"

Would be nice to have tests to include these possibilities